src/reputation/distributor.py

```diff
@@ -71,6 +72,8 @@ def reputation(self):
 
     def distribute(self):
         record = self._record_distribution()
+        self._record_referral_distribution_if_applicable(record)
```

Comment:
We should move this after line 80, so it only distributes if the original distribution is successful.

Reply:
good call!